### DISCUSSION TITLE
Monitoring 404s for the sake of SEO

### DISCUSSION THREAD

**Discussion by alexflint** [vercel/community#6625] [Help] [answered]
### Summary

I'd like to monitor 404s on my site so that I can detect URLs that are generating 404s due to removed or renamed pages on my website. Is this possible?

Just to be clear, it's not that I'm receiving some 404 errors that I don't understand, it's more that I want to see a list of all URLs that have returns a 404 in the past week, sorted by traffic volume.

### Steps to Reproduce

N/A

**Comment by amyegan** [comment, marked as answer]
Our [Monitoring](https://vercel.com/docs/observability/monitoring#) feature is an option for Pro and Enterprise customers. OTEL or Log Drains may be also be of interest. You can find more details on each of those options on the following pages:
- https://vercel.com/docs/observability/monitoring
- https://vercel.com/docs/observability/otel-overview
- https://vercel.com/docs/observability/log-drains

**Comment by versecafe** [comment]
If you follow the log drain path with something like axiom it's as simple as using the Vercel integration which will auto configure everything and then just query
```sql
vercel
| where isnotnull(['request.path']) and ['request.statusCode'] == 404
| summarize count() by bin_auto(_time), ['request.path']
```

This can quickly sort through the entire dataset and you'll get out a result like this

<img width="1359" alt="image" src="https://github.com/vercel/community/assets/147033096/6a9d8d81-633b-4075-9295-50ef4d516554">

<img width="1359" alt="image" src="https://github.com/vercel/community/assets/147033096/d4f2503b-d6b5-4b78-a7c8-8a618e148e7d">

where in this case you can see a bunch of wp-content missing since we moved to NextJS but in your case it would show the most hit 404 routes and you can limit the time range and only focus on the newer ones if desired
